Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -499,6 +499,22 @@ public void update(double[] point) {
         updateExecutor.update(point);
     }
 
+    /**
+     * Update the forest with the given point and a timestamp. The point is
+     * submitted to each sampler in the forest as if that timestamp was the correct
+     * stamp. storeSequenceIndexes must be false since the algorithm will not verify
+     * the correctness of the timestamp.
+     *
+     * @param point       The point used to update the forest.
+     * @param sequenceNum The timestamp of the corresponding point
+     */
+    public void update(double[] point, long sequenceNum) {
+        checkNotNull(point, "point must not be null");
+        checkArgument(point.length == dimensions, String.format("point.length must equal %d", dimensions));
+        checkArgument(!storeSequenceIndexesEnabled, "infeasible operation");
```

Comment:
Is this because we haven't implemented a solution to the problem of duplicate timestamps?

Reply:
Yes. This restriction should go away once HashSet -> HashMap is done. However, another restriction may appear in its place if we enable internal shingling (which we should do to ease use). If we enable both internal shingling and out of order updates then a single update of a data value may spawn multiple shingled updates -- which the executors are currently not set up to do.

Reply:
Let's give it a better error message (in this PR or a subsequent one). Maybe "This operation is not supported when sequence indexes are enabled".